### PR TITLE
Int 546 hoi backend refactor

### PR DESCRIPTION
**Pull request by SergiiRedchuk**

### CODE REVIEW

File: src/main/java/gov/ca/cwds/data/ns/ScreeningDao.java [hunk 1]
```diff
@@ -45,14 +45,15 @@ public ScreeningDao(@NsSessionFactory SessionFactory sessionFactory) {
   /**
    * Find HOI ScreeningEntity objects by screening id.
    *
-   * @param screeningId - screening Id
+   * @param clientIdList - list of Client Id-s
    * @return list of ScreeningEntity objects
    */
   @SuppressWarnings("unchecked")
-  public Set<ScreeningEntity> findHoiScreeningsByScreeningId(String screeningId) {
+  public Set<ScreeningEntity> findScreeningsByScreeningId(Set<String> clientIdList) {
```
Comment (saleemis-cwds) [inline src/main/java/gov/ca/cwds/data/ns/ScreeningDao.java]: Minor comment: The method name and parameter name do not march, should it be findHoiScreeningsForClients?

File: src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java [hunk 2]
```diff
@@ -16,28 +17,28 @@
  *
  * @author CWDS API Team
  */
-public class HOIScreeningService extends SimpleResourceService<String, HOIScreening, HOIScreeningResponse> {
+public class HOIScreeningService extends
+    SimpleResourceService<HOIScreeningRequest, HOIScreening, HOIScreeningResponse> {
 
   @Inject
-  ScreeningDao screeningDao;
+  transient ScreeningDao screeningDao;
```
Comment (saleemis-cwds) [inline src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java]: No need to make this transient

File: src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java [hunk 3]
```diff
@@ -16,28 +17,28 @@
...
 
   @Inject
-  HOIScreeningFactory hoiScreeningFactory;
+  transient HOIScreeningFactory hoiScreeningFactory;
```
Comment (saleemis-cwds) [inline src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java]: No need to make this transient

Comment (SergiiRedchuk) [reply src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java]: this is to satisfy Sonar checks

File: src/main/java/gov/ca/cwds/data/ns/ScreeningDao.java (hunk 1 above)
Comment (SergiiRedchuk) [reply src/main/java/gov/ca/cwds/data/ns/ScreeningDao.java]: renamed in https://github.com/ca-cwds/API/pull/202

File: src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java (hunk 2 above)
Comment (SergiiRedchuk) [reply src/main/java/gov/ca/cwds/rest/services/hoi/HOIScreeningService.java]: this is to satisfy Sonar checks

**Pull request merged**